code/modules/mob/mob_defines.dm

```diff
@@ -241,3 +241,5 @@
 	var/datum/vision_override/vision_type = null //Vision override datum.
 
 	var/list/permanent_huds = list()
+	
+	var/datum/smell/own_smell = null //what the mob smells like
```

Comment:
This should be defined on /living mobs, as the only other path are dead mobs (observers) and cameras, neither of which should smell.